File: problems/0088/0088-0.cpp
#include <vector>

class Solution {
public:
    void merge(std::vector<int>& nums1, int m, std::vector<int>& nums2, int n) {
        std::vector <int> res;
        res.reserve(m + n);
        int idx1 = 0;
        int idx2 = 0;
        while (idx1 != m && idx2 != n) {
            if (nums1[idx1] < nums2[idx2]) {
                res.push_back(nums1[idx1]);
                ++idx1;
            }
            else {
                res.push_back(nums2[idx2]);
                ++idx2;
            }
        }
        if (idx1 == m) {
            for (int i = idx2; i < n; ++i) {
                res.push_back(nums2[i]);
            }
        }
        else {
            for (int i = idx1; i < m; ++i) {
                res.push_back(nums1[i]);
            }
        }
    nums1 = res;
    }
};

File: problems/0001 - 0400/0088/0088-0.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        res = []
        i = 0
        j = 0
        while i < m and j < n:
            if nums1[i] < nums2[j]:
                res.append(nums1[i])
                i += 1
            else:
                res.append(nums2[j])
                j += 1

        if i == m:
            for k in range(j, n):
                res.append(nums2[k])
        else:
            for k in range(i, m):
                res.append(nums1[k])
        nums1[:] = res
